internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -372,6 +364,28 @@ public VeniceWriter(
     }
   }
 
+  private PubSubProducerAdapter getProducerAdapter() {
+    return producerAdapterLazy.get();
+  }
+
+  private ThreadPoolExecutor getThreadPoolExecutor() {
+    return threadPoolExecutorLazy.get();
+  }
```

Comment:
I think personally I would prefer not hiding usages of `Lazy` in this way. I prefer letting the `producerAdapterLazy.get()` be present at the call site, as it gives maintainers a nudge to think about whether they really want to initialize the `Lazy` instance. In some cases, it may be desirable to only use the lazy if it's already initialized, by calling `ifPresent(instance -> { ... })` on it.

Reply:
That makes sense. Will update.